src/GameLogic/ItemPriceCalculator.cs

```diff
@@ -121,55 +147,58 @@ public class ItemPriceCalculator
-                gold += 300000 * optionLevel;
-                return gold;
+                var opts = item.ItemOptions.Where(o => o.ItemOption?.OptionType == ItemOptionTypes.Option).Count();
+                return 960000 + (300000 * opts);
             }
         },
-        { (int)SpecialItems.DevilEye, item => item.Level == 1 ? 15000 : item.Level == 2 ? 21000 : (item.Level - 1) * 15000 },
-        { (int)SpecialItems.DevilKey, item => item.Level == 1 ? 15000 : item.Level == 2 ? 21000 : (item.Level - 1) * 15000 },
-        { (int)SpecialItems.DevilInvitation, item => item.Level == 1 ? 60000 : item.Level == 2 ? 84000 : (item.Level - 1) * 60000 },
-        { (int)SpecialItems.RedemyOfLove, _ => 900 },
-        { (int)SpecialItems.Rena, item => item.Level == 3 ? item.Durability() * 3900 : 9000 },
-        { (int)SpecialItems.Ale, _ => 1000 },
-        {
-            (int)SpecialItems.InvisibleCloak, item => item.Level == 1 ? 150000 :
-                item.Level == 2 ? 660000 :
-                item.Level == 3 ? 720000 :
-                item.Level == 4 ? 780000 :
-                item.Level == 5 ? 840000 :
-                item.Level == 6 ? 900000 :
-                item.Level == 7 ? 960000 :
-                item.Level == 8 ? 1200000 : 60000
-        },
         {
-            (int)SpecialItems.BloodBone, item => item.Level == 1 ? 15000 :
-                item.Level == 2 ? 21000 :
-                item.Level == 3 ? 30000 :
-                item.Level == 4 ? 39000 :
-                item.Level == 5 ? 48000 :
-                item.Level == 6 ? 60000 :
-                item.Level == 7 ? 75000 :
-                item.Level == 8 ? 90000 : 15000
+            (int)SpecialItems.DevilEye, item => item.Level == 1 ? 10000 :
+                item.Level == 2 ? 50000 :
+                item.Level == 3 ? 100000 :
+                item.Level == 4 ? 300000 :
+                item.Level == 5 ? 500000 :
+                item.Level == 6 ? 800000 :
+                item.Level == 7 ? 1000000 : 10000
         },
         {
-            (int)SpecialItems.ScrollOfArchangel, item => item.Level == 1 ? 15000 :
-                item.Level == 2 ? 21000 :
-                item.Level == 3 ? 30000 :
-                item.Level == 4 ? 39000 :
-                item.Level == 5 ? 48000 :
-                item.Level == 6 ? 60000 :
-                item.Level == 7 ? 75000 :
-                item.Level == 8 ? 90000 : 15000
+            (int)SpecialItems.DevilKey, item => item.Level == 1 ? 15000 :
+                item.Level == 2 ? 75000 :
+                item.Level == 3 ? 150000 :
+                item.Level == 4 ? 450000 :
+                item.Level == 5 ? 750000 :
+                item.Level == 6 ? 1200000 :
+                item.Level == 7 ? 1500000 : 15000
         },
+        { (int)SpecialItems.DevilInvitation, item => item.Level is 1 ? 60000 : item.Level == 2 ? 84000 : (item.Level - 1) * 60000 },    // +7 sell price on S6E3 client is 60k (same as +4). Bug?
```

Comment:
I think there is a bug in +7 DS invite price :smile: 
![Screenshot from 2024-11-27 17-59-26](https://github.com/user-attachments/assets/a67dfac5-f4e1-43f2-965e-b8aeb4ff6405)